imageio/imageio-tiff/src/main/java/com/twelvemonkeys/imageio/plugins/tiff/CCITTFaxDecoderStream.java

```diff
@@ -199,6 +206,16 @@ else if (first == 0 && (((byte) second) >> 4 == 1 || ((byte) second) == 1)) {
         return encodedType;
     }
 
+    private static byte reverse(byte x) {
+        byte b = 0;
+        for (int i = 0; i < 8; ++i) {
+            b <<= 1;
+            b |= (x & 1);
+            x >>= 1;
+        }
+        return b;
+    }
+
```

Comment:
Looks correct to me, but we already have a reverse byte lookup table in `ReverseInputStream`. It's probably faster. 😀 

Come to think of it, why don't we just wrap the stream when creating CCITT decoder streams like we do for other decoders? See `TIFFImageReader#createDecompressorStream`.

Reply:
Good idea, I moved the fillOrder handling outside to the reader.
I think the ReverseInputStream didn't exist back when I implemented the CCITT decoder 😃